src/Sushi.Mediakiwi.API/Controllers/Navigation.cs

```diff
@@ -46,14 +44,11 @@ public async Task<ActionResult<GetTopNavigationResponse>> GetTopnavigation([From
 
             var menus = await Data.Menu.SelectAllAsync().ConfigureAwait(false);
 
-            // Extract roleID from HttpContext User
-            var roleId = Utils.ConvertToInt(User.FindFirst("roleId").Value, 0);
-
             // Get Appropriate role from DB
-            var role = await Data.ApplicationRole.SelectOneAsync(roleId).ConfigureAwait(false);
+            var role = await MediakiwiUser.RoleAsync();
```

Comment:
rename RoleAsync() to SelectRoleAsync() (I assume it retrieves the role from database?)

Reply:
Agreed, adjusted and expanded comments on IApplicationUser